ta/arch/arm/link.mk

```diff
@@ -72,12 +82,21 @@ $(link-out-dir$(sm))/$(user-ta-uuid).stripped.elf: \
 	@$(cmd-echo-silent) '  OBJCOPY $$@'
 	$(q)$(OBJCOPY$(sm)) --strip-unneeded $$< $$@
 
+ifeq ($(CFG_ENCRYPT_TA),y)
+$(link-out-dir$(sm))/$(user-ta-uuid).ta: \
+			$(link-out-dir$(sm))/$(user-ta-uuid).stripped.elf \
+			$(TA_SIGN_KEY)
+	@$(cmd-echo-silent) '  SIGNENC $$@'
+	$(q)$(SIGN-ENC) --key $(TA_SIGN_KEY) --enc-key $(TA_ENC_KEY) \
+		--uuid $(user-ta-uuid) --in $$< --out $$@
+else
 $(link-out-dir$(sm))/$(user-ta-uuid).ta: \
 			$(link-out-dir$(sm))/$(user-ta-uuid).stripped.elf \
 			$(TA_SIGN_KEY)
 	@$(cmd-echo-silent) '  SIGN    $$@'
 	$(q)$(SIGN) --key $(TA_SIGN_KEY) --uuid $(user-ta-uuid) \
 		--in $$< --out $$@
+endif
```

Comment:
How about just adding an optional `--enc-key` argument to the existing `scripts/sign/py`?

Reply:
Earlier I did considered this. But since encryption TA image would add an additional header that needs to be included while computing the signature leading to parallel paths. And moreover the name of script `sign.py` maybe confusing with encryption option included. If you still think its better to maintain it in a single script, then I can extend `sign.py` as well.

Reply:
Well, the two files are still quite similar: `sign.py` is 234 lines, and you're changing -20/+40 to obtain `sign_enc.py` so effectively there are more than 200 duplicated lines (admittedly I am counting white space...).
Also with a unified script we could avoid duplicating the Makefile rule, and just pass additional flags instead.

> And moreover the name of script `sign.py` maybe confusing with encryption option included.

True, although it doesn't bother me much ;-) We can always rename the script, shouldn't be a problem.

Reply:
Ok, I will extend and rename `sign.py` script to support encryption feature. Also, will try to avoid duplicating the Makefile rule.

Reply:
> Also with a unified script we could avoid duplicating the Makefile rule

Thinking more about this. If we keep a single Makefile rule then TA build logs would provide following output irrespective of TA encryption is enabled or not which maybe confusing:

```
  SIGNENC out/b689f2a7-8adf-477a-9f99-32e90c0ad0a2.ta
```
So I think it would be better to have separate Makefile rule with following output:
Without encryption:

```
  SIGN    out/b689f2a7-8adf-477a-9f99-32e90c0ad0a2.ta
```
With encryption:

```
  SIGNENC out/b689f2a7-8adf-477a-9f99-32e90c0ad0a2.ta
```

Reply:
Thanks @jforissier for your suggestion. I have incorporated this.